sechub-cli/src/daimler.com/sechub/util/zip.go

```diff
@@ -98,20 +97,30 @@ func zipOneFolderRecursively(zipWriter *zip.Writer, folder string, zContext *zip
 		if err != nil {
 			return err
 		}
-		if zContext.filename == filePath {
+		if zContext.filename == file {
 			return errors.New(TargetZipFileLoop)
 		}
 
-		fileAbs, err := filepath.Abs(filePath)
+		fileAbs, err := filepath.Abs(file)
 		if err != nil {
 			return err
 		}
 		pwdAbs, _ := filepath.Abs(".")
 		// Make zip path relative to current working directory (the usual case)
-		zipPath := strings.TrimPrefix(fileAbs, pwdAbs+"/")
-		// If we still have a / in front: use path from folder var stripped from "./" and "../"
+		zipPath := strings.TrimPrefix(fileAbs, pwdAbs)
+
+		// tribute to Windows...
+		zipPath = strings.TrimPrefix(zipPath, filepath.VolumeName(zipPath)) // eliminate e.g. "C:"
+		zipPath = ConvertBackslashPath(zipPath)                             // '\' -> '/'
+
+		// If we still have an absolute path: use non-absolute file path stripped from "./" and "../"
 		if strings.HasPrefix(zipPath, "/") {
-			zipPath = filePath
+			zipPath = file
+
+			// Windows...
+			zipPath = strings.TrimPrefix(zipPath, filepath.VolumeName(zipPath)) // eliminate e.g. "C:"
```

Comment:
The windows specific path manipulation is done in line 113 and 114. I would prefer to have a function which can handle the path manipulation.

Furthermore, does go has functions/methods in their standard library which can handle Linux, Windows and macOS paths?

Reply:
Created a function `ConvertToUnixStylePath`